src/building/state.h

```diff
@@ -14,7 +14,8 @@
 #define BUILDING_STATE_SICKNESS (BUILDING_STATE_STRIKES + 5) // 142
 #define BUILDING_STATE_WITHOUT_RESOURCES (BUILDING_STATE_SICKNESS - RESOURCE_MAX_LEGACY) // 126 (plus variable resource size)
 #define BUILDING_STATE_DYNAMIC_RESOURCES (BUILDING_STATE_WITHOUT_RESOURCES + BUILDING_STATE_NONSTATIC_RESOURCE_SIZE)
-#define BUILDING_STATE_CURRENT_BUFFER_SIZE (BUILDING_STATE_DYNAMIC_RESOURCES + 8)
+#define BUILDING_STATE_LATRINES (BUILDING_STATE_DYNAMIC_RESOURCES + 1)
```

Comment:
This seems to be the problem responsible for the crash. The game needs to have enough space to assign all the values and then add one extra byte on top of that. So BUILDING_STATE_LATRINES should be BUILDING_STATE_DYNAMIC_RESOURCES + 9, the current value plus one. And then BUILDING_STATE_CURRENT_BUFFER_SIZE would currently be equal to BUILDING_STATE_LATRINES. (That one superflous constant is saved for the future backwards compatibility).